src/wp-settings.php

```diff
@@ -412,6 +412,124 @@
 
 $GLOBALS['wp_plugin_paths'] = array();
 
+/*
+ * Compatibility handler for for-core plugins.
+ *
+ * "For-core" are approved plugins being developed specifically to be merged into
+ * Core, e.g. Gutenberg and feature plugins.
+ *
+ * As these "for-core" plugins are frequently released, the handler identifies
+ * each's minimum compatible version, detects the version of each that is activated,
+ * and deactivates and does not load each incompatible plugin.
+ *
+ * This handler is done at the activated plugin loading loops to ensure
+ * sites stay running (to avoid fatal errors) and experiences are expected (by
+ * not loading older versions of features).
+ *
+ * @since 6.5.0
+ */
+global $_found_incompatible_for_core_plugins, $_is_plugin_compatible_with_wp, $_handle_incompatible_for_core_plugins;
+
+// Found incompatible for-core plugins.
+$_found_incompatible_for_core_plugins = array();
+
+/**
+ * Checks if the given plugin is compatible with this WordPress version.
+ *
+ * @since 6.5.0
+ *
+ * @global $_wp_for_core_plugins_compat_version
+ * @global $_found_incompatible_for_core_plugins
+ *
+ * @param string $plugin_absolute_path Absolute path to the plugin's file.
+ */
+$_is_plugin_compatible_with_wp = static function ( $plugin_absolute_path ) use ( $_wp_for_core_plugins_compat_version ) {
+	global $_found_incompatible_for_core_plugins;
+
+	static $plugins_dir_strlen;
+	if ( empty( $plugins_dir_strlen ) ) {
+		$plugins_dir_strlen = strlen( WP_PLUGIN_DIR . '/' );
+	}
+
+	$plugin = substr( $plugin_absolute_path, $plugins_dir_strlen );
```

Comment:
Why `substr()` rather than using `str_replace()`?

It's more performant. See https://3v4l.org/1BocJ